embulk-input-s3/src/main/java/org/embulk/input/s3/utils/DateUtils.java

```diff
@@ -14,8 +15,9 @@ public static Date parse(final String value, final List<String> supportedFormats
     {
         for (final String fmt : supportedFormats) {
             try {
-                return DateTimeFormat.forPattern(fmt).parseDateTime(value).toDate();
-            } catch (final IllegalArgumentException e) {
+                return new SimpleDateFormat(fmt).parse(value);
```

Comment:
Joda-Time's `DateTimeFormat` says:

> The pattern syntax is **mostly** compatible with java.text.SimpleDateFormat- **time zone names cannot be parsed** and a few more symbols are supported
https://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormat.html

So I was afraid it might make some inconpatibility in parsing timezones, but actually, this `DateUtils` is used in this plugin only with `Z` in `FULL_DATE_FORMAT` = `"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"`. So it should be okay.
* https://github.com/embulk/embulk-input-s3/blob/v0.3.5/embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java#L288
* https://github.com/embulk/embulk-input-s3/blob/v0.3.5/embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java#L55

Let me just leave a note here.

----

But because of that, `DateUtils` is used only from one, we may want to refactor `DateUtils` totally. We may remove `DateUtils`, and more straightforward direct use in `AbstractS3FileInputPlugin`. In either this or another PR.

For example, we can assume we'll have no `NullPointerException` nor `IllegalArgumentException`, but we may take care only of `ParseException`.

Reply:
Yes, I found it strange that `DateUtils` is being used here, rather just for simple 'in place' code.
Lets remove it in that PR.